applications/GeoMechanicsApplication/tests/cpp_tests/test_apply_constant_phreatic_multi_line_pressure_process.cpp

```diff
@@ -140,4 +142,45 @@ KRATOS_TEST_CASE_IN_SUITE(ApplyConstantPhreaticMultiLinePressureProcessDoesNotTh
         r_model_part, test_parameters))
 }
 
+KRATOS_TEST_CASE_IN_SUITE(ApplyConstantPhreaticMultiLinePressureProcessForShortDomain,
+                          KratosGeoMechanicsFastSuite)
+{
+    auto  model           = Model{};
+    auto& r_model_part    = model.CreateModelPart("foo");
+    r_model_part.AddNodalSolutionStepVariable(WATER_PRESSURE);
+
+    // Set up the test model part mesh
+    auto p_point_1 = Kratos::make_intrusive<Node>(1, 0.0, 0.0, 0.0);
+    auto p_point_2 = Kratos::make_intrusive<Node>(2, 0.0, 1.0, 0.0);
+    auto p_point_3 = Kratos::make_intrusive<Node>(3, 5.0, 1.0, 0.0);
+    auto p_point_4 = Kratos::make_intrusive<Node>(4, 5.0, 0.0, 0.0);
+    Quadrilateral2D4<Node> domain_geometry(p_point_1, p_point_2, p_point_3, p_point_4);
+    Parameters mesher_parameters(R"({
+        "number_of_divisions": 2,
+        "element_name": "Element2D3N",
+        "condition_name": "LineCondition",
+        "create_skin_sub_model_part": true
+    })");
+
+    StructuredMeshGeneratorProcess(domain_geometry, r_model_part, mesher_parameters).Execute();
+
+    auto  test_parameters = Parameters{R"(
+            {
+                "model_part_name": "foo",
+                "variable_name": "WATER_PRESSURE",
+                "x_coordinates": [0.0, 1.0, 2.0],
+                "y_coordinates": [1.0, 1.0, 1.0],
+                "z_coordinates": [0.0, 0.0, 0.0],
+                "gravity_direction": 1,
+                "out_of_plane_direction": 2,
+                "is_seepage": true
+            }  )"};
+
+    ApplyConstantPhreaticMultiLinePressureProcess process{r_model_part, test_parameters};
+    process.ExecuteInitialize();
+
+    KRATOS_EXPECT_DOUBLE_EQ(r_model_part.Nodes()[7].FastGetSolutionStepValue(WATER_PRESSURE, 0), -10000.0);
+    KRATOS_EXPECT_DOUBLE_EQ(r_model_part.Nodes()[8].FastGetSolutionStepValue(WATER_PRESSURE, 0), -5000.0);
```

Comment:
As far as I understand this process calculates hydrostatic water pressure for a specific coordinate based on the position of the phreatic line. @WPK4FEM , is it correct?

Reply:
That is correct.
The phreatic line is given by the x_, y_ and z_coordinates. The distance from r_model_part.Nodes()[7] or 8 to this line is computed looking in gravity_direction. The computed distance is multiplied with specific_weight.

Here specific_weight is gotten from the defaults. It is clearer when it is explicitly stated in the variable test_parameters. On the other hand, I would leave out out_of_plane_direction and is_seepage from the given parameters.

is_seepage steers an undocumented, not understood and untested piece of code. It is better not to raise the impression that we are testing is_seepage as we intend to remove all of it in one go when Jonathan has decided.

Reply:
I agree with @rfaasse that it is not obvious how the expected values have been calculated. I can imagine that we also assert the coordinates of nodes 7 and 8, because when the nodal positions change (e.g. due to changes in the mesh generator), this test may start to fail. Also, I would prefer to see how the expected water pressures are calculated. In other words, let's be explicit about the adopted density, gravity acceleration, and water depth.

Reply:
Thank you for the clarification and suggestion. `specific_weight` is added, other parameters are removed.

Reply:
As well pressure is calculated for all nodes and tested.